sqlglot/parser.py

```diff
@@ -5475,6 +5476,37 @@ def _parse_comprehension(
             condition=condition,
         )
 
+    def _parse_heredoc(self) -> exp.RawString:
+        # Note: we've already matched a $ token before reaching here
+        self._match(TokenType.PARAMETER)
+        tag_start = self._prev
+
+        while not self._match(TokenType.PARAMETER):
+            self._advance()
+            if not self._curr:
+                self.raise_error("No closing $ found.")
+
+        tag_text = self._find_sql(tag_start, self._prev)
```

Comment:
Why is there a while loop here? I think this section can be simplified, don't think we need `_find_sql` here, you can just get the token text directly from the tag's token, if there's a tag.